src/utils/deployPresearch.ts

```diff
@@ -0,0 +1,91 @@
+import type Presearch from "../types/presearch";
+import type { IProfile } from "../types/Profile";
+
+import { Network } from "../types/kubernetes";
+import createNetwork from "./createNetwork";
+
+import rootFs from "./rootFs";
+import deploy from "./deploy";
+
+const { MachinesModel, DiskModel, GridClient, MachineModel, GatewayNameModel } =
+  window.configs?.grid3_client ?? {};
+const { HTTPMessageBusClient } = window.configs?.client ?? {};
+
+export default async function deployDiscourse(
+  data: Presearch,
+  profile: IProfile
+) {
+  const name = data.name;
+  const { mnemonics, storeSecret, networkEnv } = profile;
+
+  const http = new HTTPMessageBusClient(0, "", "", "");
+  const client = new GridClient(
+    networkEnv as any,
+    mnemonics,
+    storeSecret,
+    http,
+    undefined,
+    "tfkvstore" as any
+  );
+
+  await client.connect();
+
+  await depoloyPresearchVM(data, profile);
+}
+
+async function depoloyPresearchVM(data: Presearch, profile: IProfile) {
+  const {
+    name,
+    cpu,
+    memory,
+    nodeId,
+    diskSize,
+    publicIp,
+    planetary,
+    preCode,
+    privateRestoreKey,
+    publicRestoreKey,
+  } = data;
+
+  // Private network
+  const network = createNetwork(new Network(`nw${name}`, "10.200.0.0/16"));
```

Comment:
this network's name will exceed the max length, and will result in a validation error